pytorch_lightning/loggers/wandb.py

```diff
@@ -161,11 +164,16 @@ def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) ->
         assert rank_zero_only.rank == 0, 'experiment tried to log from global_rank != 0'
 
         metrics = self._add_prefix(metrics)
-        if step is not None and step + self._step_offset < self.experiment.step:
+        if self._sync_step and step is not None and step + self._step_offset < self.experiment.step:
             self.warning_cache.warn(
-                'Trying to log at a previous step. Use `commit=False` when logging metrics manually.'
-            )
-        self.experiment.log(metrics, step=(step + self._step_offset) if step is not None else None)
+                'Trying to log at a previous step. '
+                'Use `WandbLogger(sync_step=False)` or try logging with `commit=False` when calling manually `wandb.log`.')
+        if self._sync_step:
+            self.experiment.log(metrics, step=(step + self._step_offset) if step is not None else None)
+        elif step is not None:
+            self.experiment.log({**metrics, 'trainer_step': (step + self._step_offset)})
```

Comment:
so step is the global_step from trainer, and this trainer_step is the out-of-sync step right?

Reply:
That is correct. The out-of-sync here is just due to a previous run that is resumed or a second training on a same run.

If you just have one simple training, `trainer_step == global_step` but people used to be confused on what is `global_step` in the past which is why it is renamed a bit more explicitly (+ it will avoid confusion when they are out-of-sync).

If you are talking about the wandb step, you don't see it here (which would be through `step` argument). The fact that it is not given and that there is no `commit=False` means that each batch of metrics will automatically commit and wandb step will increase at each `log`.

Once you are on your dashboard, you can decide to set your x-axis as `trainer_step` which will give you the same graph as if you had not used `sync_step=False`. The main difference is that it will handle cases where you may log multiple points per `global_step` (for example logging each validation batch, etc).

Reply:
great, that make sense. I used to log a step myself and select it in the UI. This will make it easier!